hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientMembershipListener.java

```diff
@@ -196,8 +194,7 @@ private void fireMembershipEvent(List<MembershipEvent> events) {
             if (clusterService.getMember(address) == null) {
                 Connection connection = connectionManager.getConnection(address);
                 if (connection != null) {
-                    connectionManager.destroyConnection(connection, null,
-                            newTargetDisconnectedExceptionCausedByMemberLeftEvent(member, connection.toString()));
+                    connection.close(null, newTargetDisconnectedExceptionCausedByMemberLeftEvent(member, connection.toString()));
```

Comment:
The connection.close is leading. It will do a callback to the connection manager to cleanup whatever needs to be cleaned up.